services/viva-ms/lambdas/decideCaseStatus.js

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable no-console */
+import to from 'await-to-js';
+
+import config from '../../../config';
+import * as dynamoDb from '../../../libs/dynamoDb';
+import { getStatusByType } from '../../../libs/caseStatuses';
+
+export async function main(event) {
+  const { caseKeys, workflow } = event.detail;
+
+  const newStatusType = decideNewStatusType(workflow.attributes);
+  if (newStatusType == undefined) {
+    console.info('(Viva-ms) no new status to update');
+    return true;
+  }
+
+  const [updateDbNewStatusError] = await to(
+    updateDbNewStatus(caseKeys, getStatusByType(newStatusType))
+  );
+  if (updateDbNewStatusError) {
+    console.error('(Viva-ms) updateDbNewStatusError', updateDbNewStatusError);
+  }
+
+  return true;
+}
+
+async function updateDbNewStatus(caseKeys, newStatus) {
+  const TableName = config.cases.tableName;
+
+  const params = {
+    TableName,
+    Key: caseKeys,
+    UpdateExpression: 'SET #status = :newStatusType',
+    ExpressionAttributeNames: { '#status': 'status' },
+    ExpressionAttributeValues: { ':newStatusType': newStatus },
+    ReturnValues: 'NONE',
+  };
+
+  return dynamoDb.call('update', params);
+}
+
+function decideNewStatusType(workflowAttributes) {
+  const decisionList = makeArray(workflowAttributes.decision?.decisions?.decision);
+  const paymentList = makeArray(workflowAttributes.payments?.payment);
+  const calculation = workflowAttributes.calculations?.calculation;
+
+  let decisionStatus = 0;
+  let newStatusType = '';
+
+  if (decisionList != undefined && decisionList.length > 0) {
+    decisionList.forEach(decision => {
+      decisionStatus = decisionStatus | parseInt(decision.typecode, 10);
```

Comment:
It's not clear how`decisionStatus`/`typecode` works. I assume they're some kind of flags? A suggestion would be to use named constants for these magic values.

Reply:
It's tricky yes. :)
The typeCode can be one of 01, 02 or 03.
By using the bitewise OR operator we can figure out the correct statuscode.

I'll be happy to explain in more detail IRL. :)